daml/ClaimToken/Model.daml

```diff
@@ -0,0 +1,76 @@
+module ClaimToken.Model where
+
+import DA.Finance.Types
+import DA.Finance.Asset (AssetDeposit_Transfer(..), AssetDeposit)
+import DA.Set (Set)
+
+template ClaimToken -- claim token id -> bond id
+  with
+    id: Id
+    originalId: Id
+    observers: Set Party
+  where
+    signatory id.signatories
+    observer observers
+
+    key (id.signatories, id.label): (Set Party, Text)
+    maintainer key._1
+
+    ensure 
+      id.signatories == originalId.signatories
+
+template ClaimTokenR -- reverse search (bond id -> claim token id)
```

Comment:
@georg-da  I wonder if daml allow multiple keys on the same template. If so, we don't need ClaimTokenR.

Reply:
as checked with Georg, there's no way to have multiple keys on the same template so far, will keep it as it is